neon_iris/web_sat_client.py

```diff
@@ -0,0 +1,293 @@
+                        # Process bytes message
+                        audio_bytes = message["bytes"]
+
+                        # Add extra bytes of silence if needed
+                        if len(audio_bytes) % 2 == 1:
+                            audio_bytes += b"\x00"
+
+                        # Convert audio to correct format and sample rate
+                        audio_data = np.frombuffer(audio_bytes, dtype=np.int16)
+                        if sample_rate != 16000:
+                            audio_data = resampy.resample(
+                                audio_data, sample_rate, 16000
+                            )
+
+                        # Get openWakeWord predictions and send to browser client
+                        predictions = self.oww_model.predict(audio_data)
+
+                        activations = [
+                            key for key, value in predictions.items() if value >= 0.5
+                        ]
+
+                        if activations:
+                            await websocket.send_text(
+                                json.dumps({"activations": activations})
+                            )
+
+        @self.router.post("/user_input")
+        async def on_user_input_worker(
+            req: UserInput,
+        ):
+            """
+            Callback to handle textual user input
+            @param utterance: String utterance submitted by the user
+            @returns: Session ID, audio input, audio output
+            """
+            utterance = req.utterance or ""
+            audio_input = req.audio_input or ""
+            session_id = req.session_id or "websat0000"
+
+            chat_history = []
+            input_time = time()
+            LOG.debug("Input received")
+            if not self._profiles.get("session_id"):
+                self._profiles[session_id] = {
+                    "speech": {"stt_language": self.default_lang}
+                }
+                self._current_tts[session_id] = None
+            if not self._await_response.wait(30):
+                LOG.error("Previous response not completed after 30 seconds")
+            in_queue = time() - input_time
+            self._await_response.clear()
+            self._response = None
+            self._transcribed = None
+            lang = self.get_lang(session_id)
+            if utterance:
+                LOG.info(f"Sending utterance: {utterance} with lang: {lang}")
+                self.send_utterance(
+                    utterance,
+                    lang,
+                    username=session_id,
+                    user_profiles=[self._profiles[session_id]],
+                    context={
+                        "gradio": {"session": session_id},
+                        "timing": {"wait_in_queue": in_queue, "gradio_sent": time()},
+                    },
+                )
+            else:
+                LOG.info(f"Sending audio: {audio_input} with lang: {lang}")
```

Comment:
`len(audio_input)` may be a more useful log here since normally this is a giant string of encoded audio

Reply:
Good call - I initially did want that giant string logged, so I could pull it and do manipulations with it more rapidly, but now it's just messy